### DISCUSSION TITLE
Pitchwheel on individual channels

### DISCUSSION THREAD

**Discussion by elviocip** [amsynth/amsynth#195] [Ideas]
Hi, I am a composer, I started using your great synth and I am trying to integrate it to my production line.

**The problem**: I need microtones.

**Solution #1**: I edited and loaded a Scala file for eigth-tones. 
**Why it doesn't work for me**: The Scala file loaded correctly, but unfortunately the software that I use to feed amsynth with MIDI has a different approach about microtones. Ircam's OpenMusic uses midicents to name notes (e.g. 6000 being the middle C, 6025 6050 and 6075 the microtones, 6100 being C#), detunes channels 1-4, then automatically sends the note-on messages to the right channel. I tried to send a ptchwheel message to each channel from 1 to 4, but the pitchwheel in amsynth seems to act on a global basis.

**Solution #2**: I would like to detune the channels by sending pitchwheel messages on a per-channel basis.
**Request**: would it be possible to implement this feature? 

Thanks in advance, cheers,

Elvio

**Comment by nickdowell** [comment]
Hi there!

Do you know of any other synths that support per-channel pitch wheel messages?

Taken to its logical conclusion, this sounds like multi-timbral support - i.e. making controller changes, bank/program select etc. per-channel too. This would be difficult to implement in a way that makes sense in conjunction with amsynth's UI; which channel would the UI be controlling?

**Comment by elviocip** [comment]
Thanks for your answer!
I don't know of any other synth that would let me do that because... amsynth is the first synth that I "seriously" dig.
I am, on the other hand, an old user of samplers. And any sampler using soundfonts (FluidSynth, to name one) accepts Control Change messages on a per-channel basis.
But you are right: for a synth, that would mean a multitimbral support. I guess you would need 16 instances of the class "the-whole-synth"... I ignore what would be the strain on the CPU. 
As for the UI, that wouldn't be much of a trouble: the settings will change when you change channel. The channel selection, therefore, would act as a sort of a preset selection. Having a coloured strip, with different colours associated to each different channel, would help to visually keep track, just like in any sequencer.
